File: VB-Examples/19_PivotTables/RefreshPivotTable.vb
Imports System.Data.OleDb
Imports System.Collections
Imports System.ComponentModel
Imports Spire.Xls
Imports Spire.Xls.Core.Spreadsheet
Imports Spire.Xls.Core.Spreadsheet.PivotTables

Namespace RefreshPivotTable
	Partial Public Class Form1
		Inherits Form
		Public Sub New()
			InitializeComponent()
		End Sub

		Private Sub btnRun_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btnRun.Click
            ' Create a new Workbook object
            Dim workbook As New Workbook()

            ' Load an Excel file from the specified path
            workbook.LoadFromFile("..\..\..\..\..\..\Data\Template_Xls_7.xlsx")

            ' Get the second worksheet in the workbook
            Dim sheet As Worksheet = workbook.Worksheets(1)

            ' Set the value of cell D2 in the worksheet to "999"
            sheet.Range("D2").Value = "999"

            ' Get the first PivotTable in the first worksheet and cast it to XlsPivotTable
            Dim pt As XlsPivotTable = TryCast(workbook.Worksheets(0).PivotTables(0), XlsPivotTable)

            ' Enable automatic refresh of the PivotTable cache on load
            pt.Cache.IsRefreshOnLoad = True

            ' Specify the output file name
            Dim result As String = "Result-RefreshPivotTable.xlsx"

            ' Save the modified workbook to the specified file path in Excel 2013 format
            workbook.SaveToFile(result, ExcelVersion.Version2013)

            ' Release the resources used by the workbook
            workbook.Dispose()

            'Launch the MS Excel file.
            ExcelDocViewer(result)
		End Sub

		Private Sub ExcelDocViewer(ByVal fileName As String)
			Try
				Process.Start(fileName)
			Catch
			End Try
		End Sub

		Private Sub btnClose_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btnClose.Click
			Close()
		End Sub
	End Class
End Namespace

File: CS-Examples/19_PivotTables/RefreshPivotTable.cs
using System;
using System.Data.OleDb;
using System.Drawing;
using System.Collections;
using System.ComponentModel;
using System.Windows.Forms;
using System.Data;
using Spire.Xls;
using Spire.Xls.Core.Spreadsheet;
using Spire.Xls.Core.Spreadsheet.PivotTables;

namespace RefreshPivotTable
{
	public partial class Form1 : Form
	{
        public Form1()
        {
            InitializeComponent();
        }

		private void btnRun_Click(object sender, System.EventArgs e)
		{
            // Create a workbook.
			Workbook workbook = new Workbook();

            // Load the file from disk.
            workbook.LoadFromFile(@"..\..\..\..\..\..\Data\Template_Xls_7.xlsx");

            // Get the first worksheet.
			Worksheet sheet = workbook.Worksheets[1];

            // Update the data source of PivotTable.
            sheet.Range["D2"].Value = "999";

            // Get the PivotTable that was built on the data source.
            XlsPivotTable pt = workbook.Worksheets[0].PivotTables[0] as XlsPivotTable;

            // Refresh the data of PivotTable.
            pt.Cache.IsRefreshOnLoad = true;

            // Specify the filename for the resulting file
            String result = "Result-RefreshPivotTable.xlsx";

            // Save the modified workbook to a file
            workbook.SaveToFile(result, ExcelVersion.Version2013);

            //Launch the MS Excel file.
            ExcelDocViewer(result);
		}

        private void ExcelDocViewer(string fileName)
        {
            try
            {
                System.Diagnostics.Process.Start(fileName);
            }
            catch { }
        }

        private void btnClose_Click(object sender, EventArgs e)
        {
            Close();
        }
	}
}
